cmd/geth/snapshot.go

```diff
@@ -149,6 +165,68 @@ It's also usable without snapshot enabled.
 	}
 )
 
+func pruneBlock(ctx *cli.Context) error {
+	stack, config := makeConfigNode(ctx)
+	defer stack.Close()
+
+	chaindb := utils.MakeChainDatabaseForBlockPrune(ctx, stack, false)
+	chaindb.Close()
+
+	var oldAncientPath, newAncientPath string
+	if path := getAncientPath(ctx); path != "" {
+		oldAncientPath = path + "/ancient"
+		newAncientPath = path + "/ancient_back"
+	} else {
+		utils.Fatalf("Prune failed, did not specify the AncientPath %v")
+		return errors.New("Prune failed, did not specify the AncientPath")
+	}
+
+	//TODO for filelock
+	//lock, _, err := fileutil.Flock(filepath.Join(oldAncientPath, "FLOCK"))
+	// lock, _, err := fileutil.Flock(oldAncientPath)
+	// if err != nil {
+	// 	return err
+	// }
+	// _, err = os.Open(oldAncientPath)
+	// if err != nil {
+	// 	utils.Fatalf("Failed to read genesis file: %v", err)
+	// }
+	for _, name := range []string{"chaindata"} {
+		root := stack.ResolvePath(name)
+		switch {
+		case oldAncientPath == "":
+			oldAncientPath = filepath.Join(root, "ancient")
```

Comment:
It is not possible to happen, since you forbidden it to be empty in previous logic.